sql/parse/parse.go

```diff
@@ -2484,45 +2484,61 @@ func tableExprToTable(
 		return expression.NewUnresolvedTableFunction(t.Name, exprs), nil
 
 	case *sqlparser.JoinTableExpr:
-		// TODO: add support for using, once we have proper table
-		// qualification of fields
-		if len(t.Condition.Using) > 0 {
-			return nil, sql.ErrUnsupportedFeature.New("USING clause on join")
-		}
+		return handleJoinTableExpr(ctx, t)
 
-		left, err := tableExprToTable(ctx, t.LeftExpr)
-		if err != nil {
-			return nil, err
+	case *sqlparser.ParenTableExpr:
+		if len(t.Exprs) == 1 {
+			switch j := t.Exprs[0].(type) {
+			default:
+				return nil, sql.ErrUnsupportedSyntax.New(sqlparser.String(t))
+			case *sqlparser.JoinTableExpr:
+				return handleJoinTableExpr(ctx, j)
+			}
+		} else {
+			return nil, sql.ErrUnsupportedSyntax.New(sqlparser.String(t))
 		}
+	}
+}
 
-		right, err := tableExprToTable(ctx, t.RightExpr)
-		if err != nil {
-			return nil, err
-		}
+func handleJoinTableExpr(ctx *sql.Context, t *sqlparser.JoinTableExpr) (sql.Node, error) {
```

Comment:
you might be Ok naming this `joinTableExpr`? seems like that's what most of the parse rules do